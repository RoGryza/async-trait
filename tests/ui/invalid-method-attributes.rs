use async_trait::async_trait;

#[async_trait]
pub trait Trait {
    #[async_trait]
    async fn method();
}

#[async_trait]
pub trait Trait2 {
    #[async_trait(invalid)]
    async fn method();
}

#[async_trait]
pub trait Trait3 {
    #[async_trait = "invalid"]
    async fn method();
}

fn main() {}

