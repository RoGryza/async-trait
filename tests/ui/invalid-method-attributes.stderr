error: expected attribute arguments in parentheses: #[async_trait(...)]
 --> tests/ui/invalid-method-attributes.rs:5:5
  |
5 |     #[async_trait]
  |     ^^^^^^^^^^^^^^

error: expected #[async_trait(?Send)]
  --> tests/ui/invalid-method-attributes.rs:11:5
   |
11 |     #[async_trait(invalid)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^

error: expected parentheses: #[async_trait(...)]
  --> tests/ui/invalid-method-attributes.rs:17:5
   |
17 |     #[async_trait = "invalid"]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
